app/components/show-header.js

```diff
@@ -0,0 +1,40 @@
+import Component from '@ember/component';
+import { get, set } from '@ember/object';
+
+function isAboveViewport(element) {
+  var rect = element.getBoundingClientRect();
+  return rect.top > 0;
+}
+
+let triggerStickyClass = function () {
+  let sentinel = document.querySelector(".new-donate");
+
+  if (!isAboveViewport(sentinel)) {
+    let denim = [0, 81, 142];
+    document.getElementById('sticky-donate').style.visibility = 'visible';
+    document.getElementById('listen-playhistory').style.backgroundColor = 'rgb(' + [...denim].join(',') + ')';
+  } else {
+    document.getElementById('sticky-donate').style.visibility = 'hidden';
+    document.getElementById('listen-playhistory').style.backgroundColor = 'transparent';
+  }
+};
```

Comment:
You should be able to do all this using the hooks that `ember-sticky-element` provides. You could eliminate some of this custom event code by using some of the hooks ember-sticky-element provides:

By doing something like this:
```
{{#sticky-element as |state|}}
  {{horizontal-nav
    navRoot='listen'
    defaultSlug='listen'
    channelSlug='listen'
    links=links
    class=(if state.isSticky "is-sticky")
    activeTabIndex=activeTabIndex}}
{{/sticky-element}}
```
You could just all that styling in CSS, looking for that `.is-sticky` modifier on the `horizontal-nav` class
```css
.horizontal-nav.is-sticky { 
  background-color: rgb(0,81,142);

}
```

Side note: anytime you add an event listener to the document within a component, you gotta make sure to clean it up when the element gets destroyed in that willDestroy. Using `ember-sticky-element` and its dependency `ember-in-viewport` for this sort of stuff saves the hassle of having to write all that, though.